app/models/subscription.rb

```diff
@@ -0,0 +1,35 @@
+class Subscription < ApplicationRecord
+  
+  belongs_to :owner, class_name: :User
+  belongs_to :plan
+  has_many :subscription_users
+  has_many :users, through: :subscription_users
+
+  validates :start_date, presence: true
+  validate :ensure_user_is_not_an_active_subscriber
+
+  before_create :set_expiry_date
+
+  after_create_commit :add_owner_to_users
```

Comment:
why do you want the transaction to commit before owner gets added to users. what if the callback fails?

Reply:
Changes to after_create.